node/syncblock.go

```diff
@@ -376,6 +414,48 @@ func (localNode *LocalNode) StartSyncing(syncStopHash common.Uint256, syncStopHe
 	localNode.mu.RUnlock()
 
 	syncOnce.Do(func() {
+		cs := chain.DefaultLedger.Store.(*store.ChainStore)
+
+		fastSyncHeight, err := cs.GetSyncRootHeight()
+		if err != nil {
+			log.Info("local sync root height not found:", err)
+		}
+		if config.Parameters.SyncMode != "fast" && fastSyncHeight > cs.GetHeight() {
+			err = fmt.Errorf("invalid sync mode %s at height %d", config.Parameters.SyncMode, cs.GetHeight())
+		}
+
+		if config.Parameters.SyncMode == "fast" && cs.ShouldFastSync() {
+			if fastSyncHeight == 0 {
+				fastSyncHeight = syncStopHeight - config.Parameters.RecentStateCount
```

Comment:
Need to check if `syncStopHeight > config.Parameters.RecentStateCount`. If not, we shouldn't use fast sync at all